CMakeLists.txt

```diff
@@ -34,6 +39,43 @@ if(CMAKE_C_COMPILER_VERSION VERSION_GREATER 9.1)
 	target_compile_options(rimage PRIVATE -Wno-char-subscripts)
 endif()
 
+if(${CMAKE_HOST_WIN32})
+	cmake_minimum_required(VERSION 3.20.0)
+	if(NOT OPENSSL_PATH)
+		message(FATAL_ERROR "Please set OPENSSL_PATH to your OpenSSL install directory")
+	endif()
+	cmake_path(ABSOLUTE_PATH OPENSSL_PATH NORMALIZE OUTPUT_VARIABLE OPENSSL_PATH)
+	cmake_path(APPEND OPENSSL_PATH "include" OUTPUT_VARIABLE OPENSSL_INCLUDE_PATH)
+	cmake_path(APPEND OPENSSL_PATH "lib" OUTPUT_VARIABLE OPENSSL_LIB_PATH)
+	cmake_path(APPEND OPENSSL_PATH "bin" OUTPUT_VARIABLE OPENSSL_BIN_PATH)
+
+	# If you link with OpenSSL .DLLs, then you're expected to include into
+	# your application code small "shim" snippet, which provides glue between
+	# OpenSSL BIO layer and your compiler run-time. See the OPENSSL_Applink
+	# manual page for further details.
+	target_include_directories(rimage PRIVATE "${OPENSSL_INCLUDE_PATH}")
+	target_link_directories(rimage PRIVATE "${OPENSSL_LIB_PATH}")
+	add_library(crypto SHARED IMPORTED GLOBAL)
+
+	set_property(TARGET crypto PROPERTY IMPORTED_IMPLIB_RELEASE
+		"${OPENSSL_LIB_PATH}/libcrypto.lib")
+	set_property(TARGET crypto PROPERTY IMPORTED_LOCATION_RELEASE
+		"${OPENSSL_BIN_PATH}/libcrypto-1_1-x64.dll")
+	set_property(TARGET crypto PROPERTY IMPORTED_IMPLIB_DEBUG
+		"${OPENSSL_LIB_PATH}/libcrypto.lib")
+	set_property(TARGET crypto PROPERTY IMPORTED_LOCATION_DEBUG
+		"${OPENSSL_BIN_PATH}/libcrypto-1_1-x64.dll")
+
+	target_sources(rimage PRIVATE "${OPENSSL_INCLUDE_PATH}/openssl/applink.c")
+	set_source_files_properties("${OPENSSL_INCLUDE_PATH}/openssl/applink.c" PROPERTIES
+		COMPILE_OPTIONS "-Wno-missing-prototypes")
+	target_compile_options(rimage PRIVATE -Wl,-EL)
```

Comment:
Can you explain why `-EL` is needed? rimage runs on the system that builds SOF so I wouldn't expect any endianness issue there. Keep in mind many people are not familiar with Windows around here (in case that wasn't obvious yet...)

I'm also curious what toolchain you tested this with.

Reply:
Yes we had mismatches as Windows is little-endian and we gcc we used on Windows was form MSYS2.
We use https://www.msys2.org/ which allows mixing posix tools with Windows easily.
Eg. you install MSYS2 to C:\msys64 and this separate shell establishes linux root dir there.
But they fun part is that executable binaries are windows .exe format.
So you end up with adding to Windows PATH C:\msys64\usr\bin and can easily invoke posix tools like gcc.exe.